base/compiler/typeinfer.jl

```diff
@@ -286,28 +286,34 @@ end
 function CodeInstance(result::InferenceResult, @nospecialize(inferred_result::Any),
                       valid_worlds::WorldRange)
     local const_flags::Int32
+    res = result.result
+    rettype = widenconst(res)
     if inferred_result isa Const
         # use constant calling convention
         rettype_const = (result.src::Const).val
         const_flags = 0x3
         inferred_result = nothing
     else
-        if isa(result.result, Const)
-            rettype_const = (result.result::Const).val
+        if isa(res, Const)
+            rettype_const = res.val
             const_flags = 0x2
-        elseif isconstType(result.result)
-            rettype_const = result.result.parameters[1]
+        elseif isconstType(res)
+            rettype_const = res.parameters[1]
             const_flags = 0x2
-        elseif isa(result.result, PartialStruct)
-            rettype_const = (result.result::PartialStruct).fields
+        elseif isa(res, PartialStruct)
+            rettype_const = res.fields
             const_flags = 0x2
         else
+            if isa(res, Conditional)
+                # TODO: put this into its own field ?
+                rettype = transform_to_interconditional(res, length(result.argtypes))
```

Comment:
Why not put this into `rettype_const`?

Reply:
It's just I couldn't make `InterConditional` work in `rettype_const` yesterday. Now it works.